libenv/unix_iface.c

```diff
@@ -657,9 +659,13 @@ static void FindV6InterfacesInfo(EvalContext *ctx)
 
                 if ((IsIPV6Address(ip->name)) && ((strcmp(ip->name, "::1") != 0)))
                 {
+                    char prefixed_ip[CF_MAX_IP_LEN + sizeof(IPV6_PREFIX)] = {0};
                     Log(LOG_LEVEL_VERBOSE, "Found IPv6 address %s", ip->name);
                     EvalContextAddIpAddress(ctx, ip->name, NULL); // interface unknown
                     EvalContextClassPutHard(ctx, ip->name, "inventory,attribute_name=none,source=agent");
+
+                    xsnprintf(prefixed_ip, CF_MAX_IP_LEN, IPV6_PREFIX"%s", ip->name);
```

Comment:
You forgot to update this. Better use `sizeof(prefixed_ip)`. I tend to always use `sizeof()` instead of the length directly, in order to avoid this kind of mishaps.

Reply:
Oh, that's a nice trick. it didn't occur to me that the compiler knows the size even when it's just a pointer in the end.